noir-projects/noir-contracts/contracts/token_contract/src/types/token_note.nr

```diff
@@ -52,7 +52,7 @@ impl NoteInterface<TOKEN_NOTE_LEN> for TokenNote {
     fn broadcast(self, context: &mut PrivateContext, slot: Field) {
       // We only bother inserting the note if non-empty to save funds on gas.
       if !(self.amount == U128::from_integer(0)) {
-          let encryption_pub_key = get_public_key(self.owner);
+          let encryption_pub_key = get_public_key_with_master_nullifier_public_key_hash(self.owner_nullifying_public_key_hash);
```

Comment:
We should probably add a todo in here. This should be the viewing keys, and essentially changed a lot with the incoming note changes as part of #5606